### PR TITLE
update vignettes

### PR DESCRIPTION
**Pull request by mhallal1**
part of https://github.com/insightsengineering/teal/issues/700

1) added `join_keys` and `preprocessing data` vignettes from `teal`
2) Replaced instances as `[XXX()]` with `XXX`
3) Removed the long printed datasets when possible and the unnecessary messages

### CODE REVIEW

File: vignettes/preprocessing-data.Rmd [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+author: "NEST CoreDev"
+date: "2022-07-26"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Pre-processing data}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+"Data" preprocessing refers to the code which precedes the `teal` app initialization in `app.R`, i.e.:
```
Comment (nikolas-burkoff) [inline vignettes/preprocessing-data.Rmd:12]: We should probably tweak this vignette as it refers to teal apps but we're in teal.data (we have an eval FALSE on line 21)? Other lines w.g. line 17 don't make sense in the teal.data context

File: vignettes/join-keys.Rmd [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+Refer to `vignette("including-adam-data-in-teal", package = "teal")` for more information.
+
+### Primary key
+
+Using the `keys` argument to the `dataset` (or for `DDL` a `dataset_connector`) function, we can specify column(s) of the
+dataset which together uniquely identify rows in the dataset. `teal` throws an error if it finds at least two rows
+sharing all the same values of the primary key columns.
+
+```{r, eval=FALSE}
+library(teal)
```
Comment (nikolas-burkoff) [inline vignettes/join-keys.Rmd:33]: I think we should tweak these to not use teal and some of the text may need tweaking now this has been moved

**Review by nikolas-burkoff — COMMENTED** [review]
- Delayed Data Loading vignette is for people wanting to create their own Connectors etc. and goes into crazy detail whereas Delayed Data Objects vignette is for people wanting to use DDL so maybe we should rename them and add that emphasis at the beginning of the vignettes?

File: vignettes/preprocessing-data.Rmd (hunk 1 above)
Comment (mhallal1) [reply vignettes/preprocessing-data.Rmd:12]: Updated.

File: vignettes/join-keys.Rmd (hunk 2 above)
Comment (mhallal1) [reply vignettes/join-keys.Rmd:33]: Updated.

**Review by nikolas-burkoff — APPROVED** [review]

**Pull request merged**